packages/admin/src/snackbar/SnackbarProvider.tsx

```diff
@@ -0,0 +1,67 @@
+import { SnackbarCloseReason, SnackbarProps } from "@material-ui/core";
+import * as React from "react";
+
+export interface SnackbarApi {
+    showSnackbar: (newSnackbar: React.ReactElement) => void;
```

Comment:
With this API design, I might call `api.showSnackbar(<Button {....buttonProps} />)` without problems. Isn't there a way to improve the typing? Maybe `React.ReactElement<SnackbarProps>` (haven't tried it)?

Reply:
Makes sense, since the element is cloned later in the process, it should at least support the props that get injected during `React.cloneElement`

```
                open: open,
                onClose: (event, reason) => handleClose(event, reason, snackbar?.props.onClose),
```

Reply:
Makes sense. I added the `SnackbarProps`